### PR TITLE
Fixed FK issue #1546

### PR DESCRIPTION
**Pull request by Hydrocharged**
Fixes https://github.com/dolthub/dolt/issues/1546

### CODE REVIEW

File: go/libraries/doltcore/diff/table_deltas.go [hunk 1]
```diff
@@ -257,6 +259,32 @@ func GetTableDeltas(ctx context.Context, fromRoot, toRoot *doltdb.RootValue) (de
 		})
 	}
 
+	sort.Slice(deltas, func(i, j int) bool {
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/diff/table_deltas.go:262]: It seems somewhat fragile to have the staging logic in `MoveTablesBetweenRoots` depend on the sorting order here. See comment below

File: go/libraries/doltcore/env/actions/table.go [hunk 2]
```diff
@@ -48,6 +48,7 @@ func MoveTablesBetweenRoots(ctx context.Context, tbls []string, src, dest *doltd
 			}
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/env/actions/table.go:48]: What if we just split this loop into two: one loop that processes all the drops, followed by another that processes the rest?

Comment (Hydrocharged) [reply go/libraries/doltcore/env/actions/table.go:48]: That seems fine, I'll change it.

**Review by andrew-wm-arthur — APPROVED** [review]
LGTM

**Pull request merged**